src/types/PLC/xml-data/data-types/array/data-type-array.ts

```
@@ -0,0 +1,14 @@
+import { baseTypes } from '@root/shared/data'
+import { z } from 'zod'
+
+const dataTypeArraySchema = z.object({
+  dimension: z.array(
+    z.object({
+      '@lower': z.string(),
+      '@upper': z.string(),
+    }),
+  ),
+  baseType: z.object(baseTypes.reduce((acc, type) => ({ ...acc, [type]: z.string().optional() }), {})),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)